app/hooks/useTrucoGame.ts

```diff
@@ -218,6 +243,98 @@ export function useTrucoGame() {
+      console.error('Error: Attempted to offer envido in invalid game state. This should not be possible.');
+      return;
+    }
+    
+    // Cannot offer envido when someone has flor - use original cards for accurate flor detection
+    const humanHasFlor = hasFlor(gameState.originalHumanCards, gameState.muestraCard);
+    const aiHasFlor = hasFlor(gameState.originalAiCards, gameState.muestraCard);
+    
+    if (humanHasFlor || aiHasFlor) {
+      console.error('Error: Attempted to offer envido when there is flor. This should not be possible.');
+      setGameState(prev => ({
+        ...prev,
+        message: 'No se puede cantar envido cuando hay flor.'
+      }));
+      return;
+    }
+
+    
+    // Calculate envido points
+    const humanEnvidoPoints = calculateEnvidoPoints(gameState.humanCards, gameState.muestraCard);
+    const aiEnvidoPoints = calculateEnvidoPoints(gameState.aiCards, gameState.muestraCard);
+    
+    setGameState(prev => ({
+      ...prev,
+      envidoState: { 
+        type: 'CALLED', 
+        lastCaller: 'HUMAN',
+        humanPoints: humanEnvidoPoints,
+        aiPoints: aiEnvidoPoints 
+      },
+      message: 'Gritaste envido! El jugador CPU está decidiendo...'
+    }));
+
+    const aiDecision = await getEnvidoOfferAIDecision(
+      gameState.aiCards,
+      gameState.humanCards,
+      gameState.muestraCard,
+      gameState.playedCards,
+      gameState.roundState
+    );
+
+    if (aiDecision.action === 'accept') {
+      const humanPoints = calculateEnvidoPoints(gameState.humanCards, gameState.muestraCard);
+      
+      setGameState(prev => ({
+        ...prev,
+        envidoState: { 
+          type: 'ACCEPTED', 
+          lastCaller: 'HUMAN', 
+          humanPoints: humanPoints,
+          aiPoints: aiDecision.points
+        },
+        phase: { type: 'SHOWING_ENVIDO_POINTS' },
+        message: `Jugador CPU aceptó envido! Tus puntos: ${humanPoints}, CPU puntos: ${aiDecision.points}`
+      }));
+
+      // Set a timeout to continue the game after showing envido points
+      setTimeout(() => {
+        setGameState(prev => {
+          const envidoWinner = prev.envidoState.humanPoints! > prev.envidoState.aiPoints! ? 'human' : 'ai';
+          return {
+            ...prev,
+            phase: { type: 'HUMAN_TURN' },
+            humanScore: envidoWinner === 'human' ? prev.humanScore + 2 : prev.humanScore,
+            aiScore: envidoWinner === 'ai' ? prev.aiScore + 2 : prev.aiScore,
+            message: envidoWinner === 'human' ? 
+              'Ganaste el envido! +2 puntos. Te toca jugar.' : 
+              'El jugador CPU ganó el envido! +2 puntos. Te toca jugar.'
+          };
+        });
+      }, 3000);
+    } else {
+      setGameState(prev => ({
+        ...prev,
+        envidoState: { 
+          type: 'REJECTED', 
+          lastCaller: 'HUMAN',
+          humanPoints: calculateEnvidoPoints(gameState.humanCards, gameState.muestraCard),
```

Comment:
When envido is rejected we don't need the points, right? Maybe it was the reason for having them optional in your first implementation?